qt/python/mantidqt/project/project.py

```diff
@@ -52,34 +52,34 @@ def save_as(self):
         The function that is called if the save as... button is clicked on the mainwindow
         :return: None; if the user cancels.
         """
-        path = None
-        first_pass = True
-        while first_pass or (os.path.exists(path) and not os.path.exists(path + (os.path.basename(path)
-                                                                                 + self.project_file_ext))):
-            first_pass = False
-            path = open_a_file_dialog(accept_mode=QFileDialog.AcceptSave, file_mode=QFileDialog.DirectoryOnly)
-            if path is None:
-                # Cancel close dialogs
-                return
+        path = self._save_file_dialog()
+        if path is None:
+            # Cancel close dialogs
+            return
 
-            # If the selected path is a project directory ask if overwrite is required?
-            if os.path.exists(os.path.join(path, (os.path.basename(path) + self.project_file_ext))):
-                answer = QMessageBox.question(None, "Overwrite project?",
-                                              "Would you like to overwrite the selected project?",
-                                              QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
-                if answer == QMessageBox.Yes:
-                    break
-                else:
-                    continue
+        overwriting = False
+        # If the selected path is a project directory ask if overwrite is required?
+        if os.path.exists(os.path.join(path, (os.path.basename(path) + self.project_file_ext))):
+            answer = QMessageBox.question(None, "Overwrite project?",
+                                          "Would you like to overwrite the selected project?",
+                                          QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
+            if answer == QMessageBox.No:
+                return
+            elif answer == QMessageBox.Yes:
+                overwriting = True
 
-            if os.path.exists(path) and os.listdir(path) != []:
-                QMessageBox.warning(None, "Empty directory or project required!",
-                                    "Please choose either an new directory or an already saved project", QMessageBox.Ok)
+        if not overwriting and os.path.exists(path) and os.listdir(path) != []:
+            QMessageBox.warning(None, "Empty directory or project required!",
+                                "Please choose either an new directory or an already saved project", QMessageBox.Ok)
```

Comment:
Are we missing a `return` here? I tried `Save As` and then just selected my home directory as the location. I got the above message box but it still saved anyway.